security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -335,6 +353,45 @@ func (sc *SecretCache) GenerateSecret(ctx context.Context, connectionID, resourc
 	return ns, nil
 }
 
+func (sc *SecretCache) addFileWatcher(file string, key ConnKey) {
+	// Check if this file is being already watched, if so ignore it. FileWatcher has the functionality of
+	// checking for duplicates. This check is needed here to avoid processing duplicate events for the same file.
+	sc.certMutex.Lock()
+	_, exists := sc.certs[file]
+	if !exists {
+		sc.certs[file] = struct{}{}
+	}
+	sc.certMutex.Unlock()
+	// File is not being watched, start watching now and trigger key push.
+	if !exists {
+		cacheLog.Debugf("adding watcher for file %s", file)
+		if err := sc.certWatcher.Add(file); err != nil {
+			cacheLog.Errorf("error adding watcher for file %s, Skipping watches ", file)
+			return
+		}
+		go func() {
+			var timerC <-chan time.Time
+			for {
+				select {
+				case <-timerC:
+					timerC = nil
+					if val, ok := sc.secrets.Load(key); ok {
```

Comment:
What happens if the file we are watching is no longer present in any configs?

Reply:
I think we can remove it from watcher if Load fails here?

Reply:
But the files are still being watched I think.
This code may have issues when used on the ingress gateway, which may watch many keys and certs.
Can you add a TODO here to safely remove the watchers for unused keys and certs?